### ISSUE TITLE
Crafting tables on moving crafts

### ISSUE THREAD

**Issue by TrollyLoki** [APDevTeam/Movecraft#562]
**Is your feature request related to a problem? Please describe.**
Containers like chests and furnaces and be opened and interacted with as a craft is moving (which is very cool), but the same does not extend to GUIs that aren't containers. This is very frustrating because you can access items while another player is flying the ship, but you can't craft anything.

**Describe the solution you'd like**
Extend whatever allows containers to stay open as crafts move to any GUI attached to a block.

**Comment by TylerS1066** [comment]
This would be neat, but I've got no clue how to add it or why it doesn't currently work.

**Comment by TrollyLoki** [comment]
It seems inventories that aren't associated with tile entities use something called ContainerLevelAccess which stores the location of the block that opened the inventory, I tried having the plugin update that value as the craft moves and it seems to work. So far I've only testing it with crafting tables on 1.19, but I imagine something similar will work on other versions and with other similar blocks like anvils. (See [my fork](https://github.com/TrollyLoki/Movecraft) for experimental implementation details)

**Comment by TrollyLoki** [comment]
This issue returned as of https://github.com/APDevTeam/Movecraft/commit/61c79dda8acaaf3cd2556d5c7f8915714b1583df

In that commit the code fixing this issue was removed under the assumption that it was "not needed for 1.20+" presumably due to the new Paper teleportation API. However, that API does not fix this issue. I have just tested it and this issue is present on the mentioned commit but not the commit immediately before it.
